File: src/Scripting/Perl/singleton.pl
use strict;
use warnings;
use Scalar::Util qw(refaddr);

{
    package Registry;
    my $instance;

    sub instance {
        $instance //= bless { count => 0 }, __PACKAGE__;
        return $instance;
    }

    sub increment {
        my ($self) = @_;
        $self->{count}++;
    }

    sub count {
        my ($self) = @_;
        return $self->{count};
    }
}

my $first = Registry::instance();
my $second = Registry::instance();
$first->increment();

print 'same=', refaddr($first) == refaddr($second) ? 'true' : 'false', "\n";
print 'count=', $second->count(), "\n";

File: src/Scripting/PHP/singleton.php
<?php
final class ProcessRegistry
{
    private static ?ProcessRegistry $instance = null;
    private int $count = 0;

    private function __construct() {}

    public static function instance(): ProcessRegistry
    {
        return self::$instance ??= new ProcessRegistry();
    }

    public function increment(): void { $this->count++; }
    public function count(): int { return $this->count; }
}

$first = ProcessRegistry::instance();
$second = ProcessRegistry::instance();
$first->increment();
echo 'same=' . (($first === $second) ? 'true' : 'false') . PHP_EOL;
echo 'count=' . $second->count() . PHP_EOL;
